launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,403 @@
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_XML_STRING);
+        Entity entityTwo = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_XML_STRING);
+        String id = KeyFactory.keyToString(entity.getKey());
+        String idTwo = KeyFactory.keyToString(entityTwo.getKey());
+        ds.put(entity);
+        ds.put(entityTwo);
+
+        when(request.getParameter(ID)).thenReturn(id);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+
+        RSS rss = new RSS(entity.getProperty(PODCAST_TITLE).toString(), entity.getProperty(MP3_LINK).toString());
+        TEST_XML_STRING = xmlString(rss);
+
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        assertEquals(stringWriter.toString(), "<p>" + TEST_XML_STRING + "</p>");
+    }
+
+    /**
+     * Expects that doGet() returns an error message when an entity with request id does not exist in Datastore.
+     * @return invalid link message
+     * @throws IOException
+    */
+    @Test
+    public void doGet_EntityNotFound() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_XML_STRING);
+        Entity entityTwo = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_XML_STRING);
+        String id = KeyFactory.keyToString(entity.getKey());
+        String idTwo = KeyFactory.keyToString(entityTwo.getKey());
+        ds.put(entity);
+        ds.put(entityTwo);
+
+        when(request.getParameter(ID)).thenReturn(id);
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        assertEquals("<p>Sorry. This is not a valid link.</p>", stringWriter.toString());
+    }
+
+    /**
+     * Asserts that doGet() returns an error message when there are no entities in Datastore period.
+     * TO-DO: add this test to testing file for LoginServlet (MVP)
+     * @throws IOException
+    */
+    @Test
+    public void doGet_ReturnsErrorMsgForNoEntitiesInDatastore() throws IOException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        assertEquals(stringWriter.toString(), "You have not created any RSS feeds.");
```

Comment:
if you're able to import the truth library, it's a much better API for asserting: https://truth.dev/

you can write:
assertThat(stringWriter.toString()).isEqualTo("blah");

if you stick with junit, it looks like the expected value should come first, not the actual value.
http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.Object,%20java.lang.Object)

Reply:
Truth library looks awesome- went with JUnit 4 since that was the walkthrough suggestion. I'll make the value switch!